x-pack/plugins/ml/public/jobs/new_job/utils/new_job_utils.js

```diff
@@ -40,53 +41,58 @@ export function getQueryFromSavedSearch(formConfig) {
   };
 }
 
-// create items used for searching and job creation.
-// takes the $route object to retrieve the indexPattern and savedSearch from the url
-export function createSearchItems($route) {
-  let indexPattern = $route.current.locals.indexPattern;
-  const query = {
-    query_string: {
-      analyze_wildcard: true,
-      query: '*'
-    }
-  };
+// Provider for creating the items used for searching and job creation.
+// Takes the $route object to retrieve the indexPattern and savedSearch from the url
+export function SearchItemsProvider(Private) {
+
+  const buildESQuery = Private(BuildESQueryProvider);
```

Comment:
I'm happy to see this change, this a lot better than using the `getQueryFromSavedSearch` function that I had created.
When this was originally written is wasn't possible to include `BuildESQueryProvider ` to create the query object as it was using relative links internally that caused issues. I think because we were then inside `x-pack-kibana`